src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.cs

```diff
@@ -1104,7 +1104,7 @@ private SamlSecurityToken ValidateSignature(SamlSecurityToken samlToken, string
                 throw LogExceptionMessage(new SecurityTokenSignatureKeyNotFoundException(FormatInvariant(TokenLogMessages.IDX10501, samlToken.Assertion.Signature.KeyInfo, exceptionStrings, samlToken)));
```

Comment:
We could change message IDX10501 to be specific for SAML and use KeyInfo as we to with IDX10512